React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+        80: "#ff1a1a",
+      },
+    };
+  },
+};
+```
+
+Then in `props` we have access button props, `theme` object and `colorMode`.
+
+Note:
+
+- `mode(lightMode, darkMode)(props)` function is the same as `colorMode === "dark" ? darkMode : lightMode`.
+- `mode` function is part of `chakra-ui/theme-tools` package.
+
+Theme folder structure:
+
+```bash
+src
+├── styles
+    └── theme
+        ├── index.ts # main theme endpoint
+        ├── styles.ts # global styles
+        ├── foundations # colors, typography, sizes...
+            ├── fontSizes.ts
+            └── colors.ts
+        ├── components # components styles
+            └── button.ts
+```
+
+<!-- TODO maybe detail description of each folder or list of files per folder? -->
+
+### Colors
+
+Color naming should a single value or an object with keys in range from 50 to 900.
+
+Example:
+
+```ts
+export const colors = {
+  white: "#ffffff",
+  primary: {
+    50: "#f6faff",
+    100: "#d7e9fd",
+    200: "#c0dcfc",
+    300: "#a7cefb",
+    400: "#8abdfa",
+    500: "#68aaf8",
+    600: "#3d92f6",
+    700: "#0070f3",
+    800: "#0064d8",
+    900: "#0055b9",
+  },
+};
+```
+
+### Global styles
+
+Global styles are theme-aware styles you can apply to any html element globally.
+
+They are defined in `src/styles/theme/styles.ts` file.
+
+<!-- TODO maybe what should be defined in global -->
+
+Example:
+
+```ts
+const overrides = {
+  styles: {
+    global: {
+      body: {
+        fontFamily: "body",
+        bg: "white",
+        color: "primary.200",
+      },
+      "*": {
+        boxSizing: "border-box",
+      },
+    },
+  },
+};
+```
+
+### Responsive
+
+Chakra UI supports responsive styles out of the box. Instead of manually adding @media queries and adding nested styles throughout your code, Chakra UI allows you to provide object and array values to add mobile-first responsive styles.
+
+> Under the hood `@media(min-width)` media query is used to ensure mobile-first
+
+Chakra UI default breakpoints:
+
+```ts
+export const breakpoints = {
+  sm: "30em",
+  md: "48em",
+  lg: "62em",
+  xl: "80em",
+};
+```
+
+Here's how to interpret this syntax:
+
+- `base`: From `0em` upwards
+- `md`: From `48em` upwards
+- `lg`: From `62em` upwards
+- `xl`: From `80em` upwards
+
+For responsive styles, array or object syntax can be used.
+
+```tsx
+// array syntax
+<Text fontSize={["24px", "40px", "56px"]}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+```tsx
+// object syntax
+<Text fontSize={{ base: "24px", md: "40px", lg: "56px" }}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+In case we need to skip cretan breakpoint, `null` is passed at that position in the array to avoid generating unnecessary CSS.
+
+Example:
+
+```jsx
+<Text fontSize={["24px", null, "56px"]}>Lorem Ipsum is simply dummy text</Text>
+```
+
+Array and Object syntax work for every style prop in the them specification, which means you can change most properties at a given breakpoint.
```

Comment:
What did you mean by `in the them specification`?

Reply:
theme*